website/addons/osfstorage/views.py

```diff
@@ -229,3 +229,28 @@ def osfstorage_download(file_node, payload, node_addon, **kwargs):
             osf_storage_settings.WATERBUTLER_RESOURCE: version.location[osf_storage_settings.WATERBUTLER_RESOURCE],
         },
     }
+
+@must_have_permission('write')
+@decorators.autoload_filenode(must_be='file')
+def osfstorage_add_tag(file_node, **kwargs):
```

Comment:
What if the user sends a request that bypasses the front-end? (via Postman, CURL, etc)

Don't rely on the front-end for your validation logic; the backend should always be responsible for the final check. Front-end validation is only there for the convenience of the user.